.NET/Microsoft.Recognizers.Definitions.Common/English/DateTimeDefinitions.cs

```diff
@@ -235,6 +235,7 @@ public static class DateTimeDefinitions
       public const string HolidayList3 = @"(?<holiday>(?:independence|presidents(?:')?|mlk|martin luther king( jr)?|canberra|ascension|columbus|tree( planting)?|arbor|labou?r|((international|int'?l)\s+)?workers'?|mother'?s?|father'?s?|female|women('s)?|single|teacher'?s|youth|children|girls|lovers?|earth|inauguration|groundhog|valentine'?s|baptiste|bastille|veterans(?:')?|memorial|mid[ \-]autumn|moon|spring|lantern)\s+day)";
       public const string HolidayList4 = @"(?<holiday>ramad(h)?an|ram(a)?zan|ramathan|eid al(-|\s+)adha|eid al(-|\s+)azha|eidul(-|\s+)azha|feast of the sacrifice|(islamic|arabic|hijri) new year|eid al(-|\s+)fitr|festival of breaking the fast)";
       public static readonly string HolidayRegex = $@"\b(({StrictRelativeRegex}\s+({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4}))|(({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+(of\s+)?({YearRegex}|{RelativeRegex}\s+year))?))\b";
+      public static readonly string HolidayWeekendRegex = $@"\b((({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+weekend)(\s+((of\s+)?({YearRegex}|{RelativeRegex}\s+year))))|(({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+((of\s+)?({YearRegex}|{RelativeRegex}\s+year)))?(\s+weekend))|(((the\s+)?weekend\s+of\s+)({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+((of\s+)?({YearRegex}|{RelativeRegex}\s+year)))?))\b";
```

Comment:
A better solution for perf would be to add the "weekend" term as an optional named group in HolidayRegex. This way the whole regex is only evaluated once, and a quick check for the group name in matches can be used to control code flow.

Reply:
Adding a new regex, and mixing the code with DatePeriodParser is not ideal in my view.
Couldn't HolidayExtractor/Parser produce either a date Holiday or a range Holiday? This seems a more general design that would also allow us to catch multi-day holidays.

Reply:
The year pieces could also be named groups to help with parsing, instead of running yet other regexes.

Reply:
if i am understanding the comment correctly, the adding of the weekend to the holiday would resolve it to a day and not to the actual weekend since it is a date and not a daterange, 

Also the holidays added don't all lend themselves to the holiday weekend structure (holidayList4 doesn't for sure)

Reply:
Holiday extraction/parser should be agnostic to one day or multi-day. We should try to move that way.
In any case, adding new regexes in multiple places is seldom desirable, as it negatively affects perf. Especially in big ever groing regexes like holidays.

Reply:
I agree that your suggestion could lead to better design overall; But I would like to note that the `HolidayExtractor` and the `HolidayParser` both only deal with date types. I checked the specs and attempted multiple multi-day holidays, but they all resolve to single point dates with the 'date' type (unless I am missing something). This is why it made sense to me to include holiday weekends in the dateperiodparser because they are 1. dateranges, 2. a special kind of weekend.

I'll address the regex change with groups now. I can also change the HolidayExtractor and HolidayParser to create and deal with dateranges and include the holiday-weekend logic if we still think that this change is worth doing. What do you think?

Reply:
The performance issue has been addressed, and so is the design suggestion of moving the logic to the holidayExtractor/Parser. Please check [the updates summary.](https://github.com/microsoft/Recognizers-Text/pull/2757#issuecomment-965860347)